packages/medusa/src/utils/clean-response-data.ts

```diff
@@ -1,21 +1,53 @@
 import { pick } from "lodash"
 
+// TODO: once the legacy totals decoration will be removed.
+// We will be able to only compute the totals if one of the total fields is present
+// and therefore avoid totals computation if the user don't want them to appear in the response
+// and therefore the below const will be removed
+const EXLCUDED_FIELDS = [
+  "shipping_total",
+  "discount_total",
+  "tax_total",
+  "refunded_total",
+  "total",
+  "subtotal",
+  "paid_total",
+  "refundable_amount",
+  "gift_card_total",
+  "gift_card_tax_total",
+  "item_tax_total",
+  "shipping_tax_total",
+  "refundable",
+  "original_total",
+  "original_tax_total",
+]
+
 /**
- * Filter response data to contain props specified in the fields array.
+ * Filter response data to contain props specified in the `storeAllowedProperties` or `adminAllowedProperties`.
+ * You can read more in the transformQuery middleware utility methods.
  *
  * @param data - record or an array of records in the response
  * @param fields - record props allowed in the response
  */
-function cleanResponseData<T>(data: T, fields: string[]) {
+function cleanResponseData<T extends unknown | unknown[]>(
```

Comment:
suggestion: Would be better to split this into 2 functions, 1 handling an object and the other handling an array, which reuses the function of handling an object.

This is really confusing to read.

Reply:
Yes good suggestion, ill include that to make it easier to use